types/header.go

```diff
@@ -53,6 +53,10 @@ func (h *Header) IsGenesis() bool {
 
 type Nonce [8]byte
 
+func (n Nonce) EncodeNonce(i uint64) {
+	binary.BigEndian.PutUint64(n[:], i)
+}
```

Comment:
Optional nitpick:
Not sure do we need this function anymore? In the original PR submitted by Dušan, it was used in multiple places, so that's why I've suggested extracting that logic in a new function.

Reply:
fixed: https://github.com/0xPolygon/polygon-edge/pull/1508/commits/76b225ee32a8cfad17b9a671d6fd0a61654f8f9f